src/types/types.ts

```diff
@@ -1624,6 +1624,228 @@ export interface SendPrivateTransactionOptions {
   fast: boolean;
 }
 
+/**
+ * Asset type returned when calling {@link TransactNamespace.simulateAssetChanges}.
+ * Allows you to determine if the assets approved or / and transferred are native, tokens or NFTs.
+ */
+/** The type of asset in a {@link SimulateAssetChange}*/
+export enum SimulateAssetType {
+  /**
+   * Native transfers
+   * i.e. involving the currency of the chain you are simulating on.
+   * ETH for Ethereum, MATIC for Polygon, ETH for Arbitrum.
+   */
+  NATIVE = 'NATIVE',
+
+  /** ERC20 approval or transfers. */
+  ERC20 = 'ERC20',
+
+  /** ERC721 approval or transfers. */
+  ERC721 = 'ERC721',
+
+  /** ERC1155 approval or transfers. */
+  ERC1155 = 'ERC1155',
+
+  /** Only applies to ApproveForAll.
+   * In rare cases, we are not yet able to get the token type.
+   * For ApproveForAll events - the event signature is not enough to differentiate between ERC721 and ERC1155 given both have the same signature.
+   * We will return NFT in this situation. */
+  NFT = 'NFT',
+
+  /** Approval or transfers.
+   * Special contracts that don't follow ERC 721/1155.
+   * Currently CryptoKitties and CryptoPunks. */
+  SPECIAL_NFT = 'SPECIAL_NFT'
+}
+
+/**
+ * Change type returned when calling {@link TransactNamespace.simulateAssetChanges}.
+ */
+export enum SimulateChangeType {
+  /**
+   * Approval
+   * APPROVE without token ID → approve all tokens
+   * APPROVE without amount → approve all amount
+   * APPROVE with zero amount → approval being cleared
+   */
+  APPROVE = 'APPROVE',
+
+  /** T */
+  TRANSFER = 'TRANSFER'
+}
+
+/**
+ * The error field returned in a {@link SimulateAssetChangesResponse} and
+ * {@link SimulateExecutionResponse} if the simulation failed.
+ */
+export interface SimulationError extends Record<string, any> {
+  /** The error message. */
+  message: string;
+}
+
+/**
+ * Represents an asset change from a call to
+ * {@link TransactNamespace.simulateAssetChanges}.
+ */
+
+export interface SimulateAssetChangesChange {
+  /** The type of asset from the transaction. */
+  assetType: SimulateAssetType;
+
+  /** The type of change from the transaction. */
+  changeType: SimulateChangeType;
+
+  /** The from address. */
+  from: string;
+
+  /** The to address. */
+  to: string;
+
+  /**
+   * The raw amount as an integer string. Only available on TRANSFER changes for
+   * ERC20 and NATIVE assets, or ERC721/ERC1155 disapprove changes (field set to
+   * '0').
+   */
+  rawAmount?: string;
+
+  /**
+   * The amount as an integer string. This value is calculated by applying the
+   * `decimals` field to the `rawAmount` field. Only available on TRANSFER
+   * changes for ERC20 and NATIVE assets, or ERC721/ERC1155 disapprove changes
+   * (field set to '0').
+   */
+  amount?: string;
+  /** The name of the asset transferred, if available. */
+  name?: string;
+  /** The symbol of the asset transferred if available.*/
+  symbol?: string;
+
+  /**
+   * The number of decimals used by the ERC20 token. Set to 0 otherwise.
```

Comment:
confirm with Malak - believe this could be null if we cannot fetch the number of decimals